### ISSUE TITLE
Error when runnning on test data

### ISSUE THREAD

**Issue by leavauchier** [geoflow3d/geoflow-bundle#12]
Hello, 

First, thanks for sharing this tool!

Second, I'm trying to run geoflow-bundle on the test dataset that you provide, but I'm getting an error that I don't quite understand so I'd appreciate any hint about what is causing the error.

### The context:
- I'm using Ubuntu 22.04
- I've installed geoflow-bundle from source using cmake (using the instructions in the README file). It seems to have succeeded.
- I use the following command:

```bash
lod22-reconstruct \
  --input_footprint=${data_dir}/wippolder.gpkg  \
  --input_pointcloud=${data_dir}/wippolder.las  \
  --config=${data_dir}/config.toml
```

### The logs:
```
Automatically loading flowchart /usr/local/share/geoflow/gfc-brecon/single/reconstruct.json
key not found in node parameters: skip_LoD12
key not found in node parameters: skip_LoD13
key not found in node parameters: skip_LoD22
Could not load node of type OGRLoader, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type OGRWriter, register not found: io-gdal
Could not load node of type PolygonBufferGEOS, register not found: io-gdal
Could not load node of type PolygonSimplifyGEOS, register not found: io-gdal
Could not connect output global_elevation_50p
Could not connect output global_elevation_70p
Could not connect output global_elevation_max
Could not connect output global_elevation_min
Could not connect output attributes
Could not connect output linear_rings
Could not connect output attributes
Could not connect output linear_rings
Could not connect output attributes
Could not connect output linear_rings
Could not connect output data_coverage
Could not connect output roof_pt_cnt
Could not connect output roof_type
Could not connect output ground_elevations
Could not connect output mesh_error_f
Could not connect output mesh_error_f
Could not connect output mesh_error_f
Could not connect output multi_triangle_collections
Could not connect output multi_triangle_collections
Could not connect output multi_triangle_collections
Could not connect output replace
Could not connect output roof_type
Could not connect output skip
Could not connect output errors
Could not connect output errors
Could not connect output errors
Reading configuration from file /path/to/data/config.toml
set global input_footprint_select = 47 (from config file)
set global input_footprint =/path/to/data//wippolder.gpkg (from command line)
set global input_pointcloud = /path/to/data//wippolder.las (from command line)
```

I've checked  ` /usr/local/share/geoflow/gfc-brecon/single/reconstruct.json` and it contains a field named parameters, that contains `skip_LoD12` etc.
Also I'm not sure of what the other errors mean.

Thanks for any advice on what is going wrong!

EDIT: remove local filepaths in log

**Comment by balazsdukai** [comment]
Hi, could you show the output of `lod22-reconstruct --list-plugins` and `lod22-reconstruct --list-nodes` ?

**Comment by leavauchier** [comment]
`lod22-reconstruct --list-plugins` returns
```
Automatically loading flowchart /usr/local/share/geoflow/gfc-brecon/single/reconstruct.json
GF_PLUGIN_FOLDER = /usr/local/lib/geoflow-plugins
 > io-las 0.2.0
   compiled with GNU v11.3.0
 > Val3dity 0.2.0
   compiled with GNU v11.3.0
 > CoreIO 0.2.0
   compiled with GNU v11.3.0
 > building-reconstruction 0.2.0
   compiled with GNU v11.3.0
 > Core [built-in]
```

and `lod22-reconstruct --list-nodes` returns 
```
Automatically loading flowchart /usr/local/share/geoflow/gfc-brecon/single/reconstruct.json
Available nodes for plugin io-las:
 > LASLoader
 > LASVecLoader
 > LASVecWriter
 > LASWriter
 > PointCloudClassSplit
Available nodes for plugin Val3dity:
 > Validator
Available nodes for plugin CoreIO:
 > CityJSONFeatureWriter
 > CityJSONWriter
 > OBJVecWriter
 > OBJWriter
 > PLYWriter
Available nodes for plugin building-reconstruction:
 > AlphaShape
 > Arr2LinearRings
 > Arr2LinearRingsDebug
 > ArrDissolve
 > ArrExtruder
 > AttrRingsSelector
 > AttributeTester
 > BuildArrFromLines
 > BuildingRasterise
 > BuildingSelector
 > CityGMLMeshWriter
 > DataCoverageCalc
 > DetectLines
 > DetectPlanes
 > FacesSelector
 > Filter25D
 > LASInPolygons
 > LOD1Extruder
 > LinearRingtoRings
 > OptimiseArrangment
 > OptimiseArrangmentGrid
 > PC2MeshQuality
 > PC2PCDistancesCalculator
 > PCFilter
 > PCRasterise
 > PlaneIntersect
 > PlaneIntersectAll
 > PointCloudMerger
 > PolygonExtruder
 > PolygonGrower
 > PolygonOffsetter
 > PolygonTriangulator
 > PolygonUnion
 > PrintResult
 > RasterMerger
 > RegulariseLines
 > RegulariseRings
 > Ring2Segments
 > RoofPartitionRasterise
 > SegmentExtend
 > SegmentRasterise
 > SimplifyPolygon
 > SkipReplaceTester
 > SnapRound
 > TriSnap
Available nodes for plugin Core:
 > Bool
 > Float
 > Int
 > NestedFlowchart
 > Text
 > TextWriter
```

**Comment by balazsdukai** [comment]
Thanks. The problem is that the gdla plugin is missing from the installation (it is not loaded). This usually happens when the gdal plugin cannot link to the gdal library.

Do you have gdal installed? If not, you need to install it first, then recompile.

In case you had gdal installed, and you did `cmake --install` (or `make install`). Unfortunately, cmake install often breaks the link to the gdal library. To verify that this happened, go to the build directory where you built geoflow, locate the gdal plugind that you've built (`gfp_gdal.so`) and do `ldd gfp_gdal.so`. You should see a very long list of libraries, each pointing to a location on your system. Most importantly, the gdal library needs to be there (in my case `libgdal.so.29 => /opt/lib/libgdal.so.29 (0x00007f52e4b96000)`).
If you repeat the same on the gdal plugin that *have been installed with cmake install*, and you see a shorter list of linked libraries, or the gdal library points to nothing, then cmake install indeed broke the link. You'll find the installed plugins at `/usr/local/lib/geoflow-plugins` (see the output of `--list-plugins`).
So, if the links have been broken by the installation, then you need to manually `cp` the built plugins from the build directory to the installation directory.

**Comment by leavauchier** [comment]
Thanks for your help! 
It looks like the installation broke the links. 
Copying the plugins seems to have solved the problem, however it looks like I had to run `cmake --build . --target install` for `./geoflow-bundle/build/geoflow/apps/geof` to run without a Segmentation fault (not only for the installed `/usr/local/bin/geof`).

In case someone else bumps into the same issue, I also changed gdal from one installed wirh `conda` to one installed with `apt`
